src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -685,3 +685,30 @@ XBUtilities::str_to_reset_obj(const std::string& str)
     return it->second;
   throw xrt_core::error(str + " is invalid. Please specify a valid reset type");
 }
+
+void
+XBUtilities::pretty_print_xrt_version()
+{
+  try {
+    boost::property_tree::ptree pt_xrt;
+    xrt_core::get_xrt_info(pt_xrt);
+    boost::property_tree::ptree empty_ptree;
+
+    std::cout << boost::format("%-20s : %s\n") % "Version" % pt_xrt.get<std::string>("version", "N/A");
+    std::cout << boost::format("%-20s : %s\n") % "Branch" % pt_xrt.get<std::string>("branch", "N/A");
+    std::cout << boost::format("%-20s : %s\n") % "Hash" % pt_xrt.get<std::string>("hash", "N/A");
+    std::cout << boost::format("%-20s : %s\n") % "Hash Date" % pt_xrt.get<std::string>("build_date", "N/A");
+    const boost::property_tree::ptree& available_drivers = pt_xrt.get_child("drivers", empty_ptree);
+    for(auto& drv : available_drivers) {
+      const boost::property_tree::ptree& driver = drv.second;
+      std::string drv_name = driver.get<std::string>("name", "N/A");
+      boost::algorithm::to_upper(drv_name);
+      std::cout << boost::format("%-20s : %s, %s\n") % drv_name
+          % driver.get<std::string>("version", "N/A") % driver.get<std::string>("hash", "N/A");
+    }
+  }
+  catch (const boost::property_tree::ptree_error &ex) {
```

Comment:
Hmm...what ptree exceptions are you expecting to catch?

I ask for it looks like you have all of the "default" values (non-existing) keys accounted for.